uint32 freq
---
bool ok
